### PR TITLE
Issue #4843: TreeWalkerFilter needs to be added to ModuleReflectionUtils

### PR DESCRIPTION
**Pull request by timurt**
https://github.com/checkstyle/checkstyle/issues/4843

### CODE REVIEW

**Comment by rnveach** [comment]
@timurt Did you confirm that `CheckUtil.getCheckstyleModules` returns all our current TreeWalkerFilters?

File: src/main/java/com/puppycrawl/tools/checkstyle/utils/ModuleReflectionUtils.java [hunk 1]
```diff
@@ -74,7 +75,8 @@ public static boolean isCheckstyleModule(Class<?> clazz) {
                     || isFileSetModule(clazz)
                     || isFilterModule(clazz)
                     || isFileFilterModule(clazz)
-                    || isRootModule(clazz));
+                    || isRootModule(clazz)
+                    || isTreeWalkerFilterModule(clazz));
```
Comment (rnveach) [inline src/main/java/com/puppycrawl/tools/checkstyle/utils/ModuleReflectionUtils.java]: Please move this closer to the other 2 filters.
Also please update javadoc to include the new module.

File: src/test/java/com/puppycrawl/tools/checkstyle/utils/ModuleReflectionUtilsTest.java [hunk 2]
```diff
@@ -53,6 +55,7 @@ public void testIsCheckstyleModule() {
         assertTrue(ModuleReflectionUtils.isCheckstyleModule(FilterClass.class));
         assertTrue(ModuleReflectionUtils.isCheckstyleModule(FileFilterModuleClass.class));
         assertTrue(ModuleReflectionUtils.isCheckstyleModule(RootModuleClass.class));
+        assertTrue(ModuleReflectionUtils.isCheckstyleModule(TreeWalkerFilterClass.class));
```
Comment (rnveach) [inline src/test/java/com/puppycrawl/tools/checkstyle/utils/ModuleReflectionUtilsTest.java]: move this closer to other filters.

Comment (timurt) [inline src/test/java/com/puppycrawl/tools/checkstyle/utils/ModuleReflectionUtilsTest.java]: done

File: src/main/java/com/puppycrawl/tools/checkstyle/utils/ModuleReflectionUtils.java (hunk 1 above)
Comment (timurt) [inline src/main/java/com/puppycrawl/tools/checkstyle/utils/ModuleReflectionUtils.java]: done

**Comment by timurt** [comment]
@rnveach 

> @timurt Did you confirm that CheckUtil.getCheckstyleModules returns all our current TreeWalkerFilters?

Yes, classes `SuppressWithNearbyCommentFilter` and `SuppressionCommentFilter` are included

**Comment by rnveach** [comment]
Thanks.
Please rebase on latest master as there is a conflict.

File: src/main/java/com/puppycrawl/tools/checkstyle/utils/ModuleReflectionUtils.java [hunk 3]
```diff
@@ -74,6 +75,7 @@ public static boolean isCheckstyleModule(Class<?> clazz) {
                     || isFileSetModule(clazz)
                     || isFilterModule(clazz)
                     || isFileFilterModule(clazz)
+                    || isTreeWalkerFilterModule(clazz)
```
Comment (rnveach) [inline src/main/java/com/puppycrawl/tools/checkstyle/utils/ModuleReflectionUtils.java]: https://github.com/checkstyle/checkstyle/pull/4854#discussion_r130225527
> Also please update javadoc to include the new module.

This point was not done.

Comment (timurt) [inline src/main/java/com/puppycrawl/tools/checkstyle/utils/ModuleReflectionUtils.java]: Sorry, done

**Pull request merged**